### ISSUE TITLE
Missing files from results

### ISSUE THREAD

**Issue by gogoprog** [ctrlpvim/ctrlp.vim#405]
Hi,

I have a custom user_command:
```vim
let g:ctrlp_user_command = 'find -L %s -type f  -not -path "*/.git/*" -path "*.hx" -o -path "*.as"'
```
When I use it manually in the terminal I can find a specific file.
But this file never appears in ctrlp.vim...

I'm not reaching the max_files limit.

Is there a known issue?

**Comment by mattn** [comment]
Did you try `%s` with root of your project directory?

**Comment by gogoprog** [comment]
Yes I always do ctrlp from my project root. The found relative filepaths match the root path.
I don't know what else to check

**Comment by mattn** [comment]
What OS do you use? And could you please try below?

```vim
:echo system(printf('find -L %s -type f  -not -path "*/.git/*" -path "*.hx" -o -path "*.as"', '/path/to/your/project/directory'))
```

**Comment by gogoprog** [comment]
I'm on Arch Linux and VIM 8.0.

Using your command I can find my file in the displayed list

**Comment by gogoprog** [comment]
Maybe some useful information:
The file I can't find is ```Image.hx``` but I can find ```ImageType.hx``` which is in the same folder...

**Comment by mattn** [comment]
Do you set `wildignore` option?

**Comment by gogoprog** [comment]
I wasn't ignoring file.

I finally found my file using g:ctrlp_max_height. So actually I just had too many matches.
How can I sort the matches in a better way?

**Comment by mattn** [comment]
How about this?

```
let g:ctrlp_path_sort = 1
let g:ctrlp_path_nolim = 1
```

**Comment by gogoprog** [comment]
It is sorted but the file I want is still way too far:

![2017-10-11_20 19 34](https://user-images.githubusercontent.com/1249453/31458951-e4d91820-aec1-11e7-9a51-bed8b670e2cb.png)
In this sample the one I want is the top one and I have to increase the max_height to see it.

Any other idea? Other fuzzy finders as the one in Visual Studio Code seem to show first the exact filename match

Thanks for the help ;)

**Comment by gogoprog** [comment]
Actually something like "the ones with less separated searched strings first"

**Comment by anowlcalledjosh** [comment]
Something like that was also suggested [two years ago](https://github.com/ctrlpvim/ctrlp.vim/issues/69#issuecomment-75499349), but doesn't seem to be implemented. If I search in my project for `run`, I expect to get `project/blueprints/run/__init__.py` selected first, but instead I get `project/blueprints/__init__.py`.

![image](https://user-images.githubusercontent.com/9433472/31963277-06f30e62-b8f8-11e7-9ce4-00b26c3c0323.png)

Obviously in this example I can just press up once to select the correct file, but in the past I have been able to press enter without even looking at CtrlP's suggestions and still get the correct file; it's a shame not to be able to do this any more.

**Comment by gogoprog** [comment]
I'll take a look to see if I can implement it.
The sort by a score computed from the separations count seems nice to me

**Comment by gogoprog** [comment]
I'm using `ctrlp_match_natural_name` at this moment it gives better results (not perfect yet tho)
